lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -272,6 +270,34 @@ bool isTppZero(linalg::GenericOp linalgOp, SmallVectorImpl<Value> *operands) {
   return true;
 }
 
+void splitFusedOp(tpp::FusedBrgemmOp fusedBrgemmOp, PatternRewriter &rewriter) {
+  assert(fusedBrgemmOp.hasBufferSemantics() &&
+         "tpp.fused_brgemm expects a memref type");
+
+  Location loc = fusedBrgemmOp.getLoc();
+
+  // Split the fused op into individual operations.
+  auto ins = fusedBrgemmOp.getInputs();
+  auto out = fusedBrgemmOp.getOutput();
+  rewriter.create<tpp::BrgemmOp>(loc, ValueRange{ins[0], ins[1], ins[2]}, out);
```

Comment:
You need to replace or you'll end up with an extra op?

Reply:
I let the caller handle the cleanup.

Reply:
Hmm I think I'm fine with either approach. Do you think erasing the original op within `splitFusedOp` is more intuitive?

TODO: the function should return a SmallVector with created ops

Reply:
TODO done

Reply:
Reworked the function